tests/test_s3urlcache.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+import logging
+import os
+import io
+import shutil
+import sys
+import unittest
+import threading
+import requests
+from http.server import BaseHTTPRequestHandler, HTTPServer
+
+pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))  # noqa
+sys.path.insert(0, pkg_root)  # noqa
+
+from dss import Config, BucketConfig
+from dss.config import Replica
+from tests.infra import start_verbose_logging, testmode
+from dss.storage.validator import S3UrlCache
+from dss.storage.validator import SizeLimitError
+from dss.util import networking
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+start_verbose_logging()
+
+KB = 1024
+MB = KB**2
+
+RANDOMDATA = os.urandom(MB)
```

Comment:
This is evaluated when the module is first imported. Typically you want to avoid interacting with the operating system at module load time. So I think this should be initialized in `setupModule()`. And since its not exactly a constant, its name should be lower case.

Reply:
I'll make the change.

> Typically you want to avoid interacting with the operating system at module load time. 

Do you have an example on why this bad?

Reply:
Good question. Consider a module `foo.py` with 

```
import os
os.system('rm -rf /')
def f():
    …
```

Simply importing `foo` will trigger the side-effect. Granted, this is an extreme (and malicious) side-effect, but it illustrates that the importing module has no control over the side-effects in the imported module, and no way to opt out. If all side-effects are wrapped in functions or classes, they lose the "side-" aspect and the importing module is given back control over what to invoke. The damage this anti-pattern can do depends on how drastic the side-effects are. `os.urandom(1MB)` isn't the end of the world. But there used to be a time where Windows didn't have a `os.urandom` implementation, so a user would not have been able to import your module at all on Windows. 

This is the same reason that Python scripts (and modules invoked via python -m) should have `if __name__ == `__main__:` guarding the module-level statements. It allows the script to be run from the command line AND to be imported as a module. When imported, the `if` statement prevents the execution of the module-level statements.

I'm not alone:

https://news.ycombinator.com/item?id=7536246